core-rust/mesh-api-server/src/mesh_api/conversions/block.rs

```diff
@@ -0,0 +1,71 @@
+use crate::prelude::*;
+
+pub fn extract_state_version_from_mesh_api_partial_block_identifier(
+    block_identifier: &models::PartialBlockIdentifier,
+) -> Result<Option<StateVersion>, ExtractionError> {
+    let state_version = match &block_identifier.hash {
+        Some(hash) => {
+            let index_from_hash =
+                hash.parse::<i64>()
```

Comment:
As mentioned yesterday, it woudl be nice to create our own block hash by concatenating parts of the transaction tree hash, receipt hash, and state version.

On read, we can resolve from the state version, and verify the other hash parts line up.

Reply:
Done